pkg/auth/dependency/oauth/handler/result_authz.go

```diff
@@ -84,3 +93,16 @@ func (a authorizationResultRequireAuthn) WriteResponse(rw http.ResponseWriter, r
 func (a authorizationResultRequireAuthn) IsInternalError() bool {
 	return false
 }
+
+func redirect(rw http.ResponseWriter, redirectURI string) {
```

Comment:
We can keep the 302 response code here for better redirection in normal cases.

Reply:
The tricky part of ios bug is if the response is 302, `ASWebAuthenticationSession` won't redirect and render the page :(

Reply:
So not only it won't redirect, it also not render the page? That's a really bad bug :(

Reply:
Just check on the device again, the page will be rendered. Update the PR, thanks @kiootic